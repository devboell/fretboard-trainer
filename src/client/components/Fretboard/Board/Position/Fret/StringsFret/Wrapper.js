import styled, { css } from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  ${props => props.isHighlighted && css`
    height:100%;
    width: auto;
    -moz-border-radius: 50%; 
    -webkit-border-radius: 50%; 
    border-radius: 50%;
    background: #4679BD; 
  `}
`
//     padding-bottom: 50%;

//     background-color: ${props.theme.highlight};


export default Wrapper
