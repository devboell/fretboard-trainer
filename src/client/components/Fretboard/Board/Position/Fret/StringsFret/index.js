import React from 'react'
import pt from 'prop-types'
import StringsContent from '../StringsContent'
import Wrapper from './Wrapper'


const StringsFret = ({
  note, showNotes, isHighlighted,
}) =>
  <Wrapper {...{ isHighlighted }}>
    {showNotes &&
      <StringsContent {...{ note, isHighlighted }} />
    }
  </Wrapper>

StringsFret.propTypes = {
  note: pt.string.isRequired,
  showNotes: pt.bool.isRequired,
  isHighlighted: pt.bool.isRequired,
}

export default StringsFret
