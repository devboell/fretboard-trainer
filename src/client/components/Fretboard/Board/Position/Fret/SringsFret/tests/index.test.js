import React from 'react'

import StringsFret from '../index'

it('StringsFret component, snapshot', () => {
  const wrapper = shallow(<StringsFret note="E2" />)
  expect(wrapper).toMatchSnapshot()
})
