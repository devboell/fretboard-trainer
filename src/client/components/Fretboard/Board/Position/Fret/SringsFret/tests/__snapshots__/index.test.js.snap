// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringsFret component, snapshot 1`] = `
<styled.div>
  E2
</styled.div>
`;
